### PR TITLE
Support run  container with network id

### PR DESCRIPTION
**Pull request by fahedouch**
fix https://github.com/containerd/nerdctl/issues/2436

### CODE REVIEW

**Review by yankay — APPROVED** [review]
Thanks @fahedouch

File: cmd/nerdctl/container_run_network_linux_test.go [hunk 1]
```diff
@@ -49,12 +51,24 @@ func TestRunInternetConnectivity(t *testing.T) {
...
+		},
 		{
 			args: []string{"--net", "host"},
 		},
 	}
 	for _, tc := range testCases {
 		tc := tc // IMPORTANT
+		if tc.dockerIncompatible {
+			// docker compute id algorithm is different from nerdctl
+			testutil.DockerIncompatible(t)
```
Comment (AkihiroSuda) [inline cmd/nerdctl/container_run_network_linux_test.go:70]: This should be compatible if you inspect the id via `docker network inspect`

Comment (fahedouch) [reply cmd/nerdctl/container_run_network_linux_test.go]: fixed

**Comment by AkihiroSuda** [comment]
Please rebase to make CI green

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**